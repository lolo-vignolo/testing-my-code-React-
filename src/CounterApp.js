import React, {useState} from 'react';
import PropTypes from "prop-types";


const CounterApp = ({value = 10})=>{
    
   const [counter, setCounter] = useState(value);


   const handleAdd = ()=>{
        setCounter(counter +1);
       //si no tengo el valor de useState dev poner dentro del setteador una nueva funcion
       //donde c (cualquier valor) tomara el lugar del useState faltante.
      // setCounter((c)=> c + 1 );
   }

   const handleSust = ()=>{
    setCounter(counter - 1);
   }

   const resetCount = ()=>{
    setCounter(value);
    }



    return (
        <>
            <h1>CounterApp</h1> 
            <p> {counter} </p>

            <button onClick= {handleAdd}>+1</button>
            <button onClick = {resetCount} >Reset </button>
            <button onClick= {handleSust } > -1 </button>
        </>
    );
}

CounterApp.propTypes = {
    value : PropTypes.number, 

}

export default CounterApp;