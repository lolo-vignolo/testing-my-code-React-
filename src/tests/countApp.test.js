import React from "react";

import { shallow } from "enzyme"
import CounterApp from "../CounterApp.js";

describe("testisn my counter App", ()=>{

    let wrapper = shallow(<CounterApp />);

    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    });

    
    test('que se renderice el valor inicial si no asigno ninguno ', () => {

       
        expect (wrapper).toMatchSnapshot();
    });
    

    test('debarÃ­a haber un valor por defecto de 0', () => {
        
        const valorInicial = "100";

        const wrapper = shallow(<CounterApp value={100} />);
        const number = wrapper.find("p").text().trim();

        expect(number).toBe(valorInicial);
    });
    

    test('incrementar uno con c/click', () => {

        // const wapper = shallow(<CounterApp />); al sacarlo afuera
        // no hace falta que lo escriba de nuevo aqui.

        const bt1 = wrapper.find("button").at(0).simulate("click");
        const number = wrapper.find("p").text().trim();

        expect(number).toBe("11");
        
    });

    test('decrecer uno con c/click', () => {

        // const wapper = shallow(<CounterApp />); al sacarlo afuera
        // no hace falta que lo escriba de nuevo aqui.

        const bt1 = wrapper.find("button").at(2).simulate("click");
        const number = wrapper.find("p").text().trim();

        expect(number).toBe("9");
        
    }) ;
    

    test('reiniciar al valor por defecto', () => {

        // const wapper = shallow(<CounterApp />); al sacarlo afuera
        // no hace falta que lo escriba de nuevo aqui.
        const wrapper = shallow(<CounterApp value={105} />);

        const bt0 = wrapper.find("button").at(0).simulate("click");
        
        const bt1 = wrapper.find("button").at(1).simulate("click");
        const number = wrapper.find("p").text().trim();

        expect(number).toBe("105");

       

        
        
    }) ;
    
    

})