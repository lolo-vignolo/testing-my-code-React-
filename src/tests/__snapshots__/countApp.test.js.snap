// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testisn my counter App que se renderice el valor inicial si no asigno ninguno  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
     
    10
     
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
